src/app/store/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { MatSnackBar } from '@angular/material';
+
+import { Product } from '../../core/store/product.model';
+import { StoreService } from '../../core/store/store.service';
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss']
+})
+export class ProductDetailComponent implements OnInit {
+  product: Product = new Product();
+  shoppingForm: FormGroup;
+  errors: any[] = [];
+  loading = false;
+
+  constructor(
+    private storeService: StoreService,
+    private formBuilder: FormBuilder,
+    private route: ActivatedRoute,
+    private router: Router,
+    private snackBar: MatSnackBar,
+  ) {
+  }
+
+  ngOnInit() {
+    this.initializeForm();
+    this.storeService.getProduct(this.route.snapshot.params['id']).subscribe(result => {
+      this.product = result;
+      if (this.product.sizes && this.product.sizes.length === 0) {
+        this.shoppingForm.get('size').clearValidators();
+        this.shoppingForm.get('size').updateValueAndValidity();
+      } else {
+        this.shoppingForm.get('size').setValidators(Validators.required);
+      }
+    })
```

Comment:
should be consistent with end of line use of semicolons

Reply:
agree